pykickstart/commands/bootc.py

```diff
@@ -0,0 +1,96 @@
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pykickstart.version import F43
+from pykickstart.base import KickstartCommand
+from pykickstart.options import KSOptionParser
+
+class F43_Bootc(KickstartCommand):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+    conflictingCommands = ["ostreecontainer", "ostreesetup"]
+
+    def __init__(self, *args, **kwargs):
+        KickstartCommand.__init__(self, *args, **kwargs)
+
+        self.op = self._getParser()
+
+        self.stateroot = kwargs.get('stateroot', None)
+        self.sourceImgRef = kwargs.get("source-imgref", None)
+        self.targetImgRef = kwargs.get("target-imgref", self.sourceImgRef)
+
+    def __str__(self):
+        retval = KickstartCommand.__str__(self)
+
+        if not self.seen:
+            return retval
+
+        retval += "# Bootc setup\n"
+        retval += "bootc %s\n" % self._getArgsAsStr()
+
+        return retval
+
+    def _getArgsAsStr(self):
+        retcmd = []
+
+        if self.stateroot:
+            retcmd.append('--stateroot="%s"' % self.stateroot)
+        if self.sourceImgRef:
+            retcmd.append('--source-imgref="%s"' % self.sourceImgRef)
+        if self.targetImgRef:
+            retcmd.append('--target-imgref="%s"' % self.targetImgRef)
+
+        return ' '.join(retcmd)
+
+    def _getParser(self):
+        op = KSOptionParser(prog="bootc", description="""
+                            Used for Bootc installations from native container. See
+                            https://bootc-dev.github.io/bootc//man/bootc-install-to-filesystem.html
+                            for more information about Bootc install to filesystem.
+
+                            **Experimental. Use on your own risk.**
+                            """, version=F43, conflicts=self.conflictingCommands)
+        # Rename the osname to stateroot and set default as proposed by
+        # https://github.com/ostreedev/ostree/issues/2794
+        op.add_argument("--stateroot",
+                        version=F43,
+                        help="""
+                        Name for the state directory, also known as "osname".
+                        """)
+        op.add_argument("--source-imgref",
+                        required=True,
+                        version=F43,
+                        help="""
+                        Install the system from an explicitly given source.
+                        """)
+        op.add_argument("--target-imgref",
+                        version=F43,
+                        help="""
+                        Specify the image to fetch for subsequent updates.
```

Comment:
Please add the note that default is using `--source-imgref`.

Reply:
Done